modules/apps/headless-apio/blog/blog-apio-client-test/src/testIntegration/java/com/liferay/blog/apio/client/test/BlogApioTest.java

```diff
@@ -0,0 +1,328 @@
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+/**
+ * @author Víctor Galán
+ */
+@RunAsClient
+@RunWith(Arquillian.class)
+public class BlogApioTest {
+
+	@Deployment
+	public static Archive<?> getArchive() throws Exception {
+		return OAuth2ProviderTestUtil.getArchive(
+			BlogApioTestBundleActivator.class);
+	}
+
+	@Before
+	public void setUp() throws MalformedURLException {
+		_rootEndpointURL = new URL(_url, "/o/api");
+
+		String contentSpaceHref = ContentSpaceApioTestUtil.getContentSpaceHref(
+			_rootEndpointURL.toExternalForm(),
+			BlogApioTestBundleActivator.CONTENT_SPACE_NAME);
+
+		_blogPostsHref = ApioClientBuilder.given(
+		).basicAuth(
+			"test@liferay.com", "test"
+		).header(
+			"Accept", "application/hal+json"
+		).when(
+		).get(
+			contentSpaceHref
+		).then(
+		).extract(
+		).path(
+			"_links.blogPosts.href"
+		);
+	}
+
+	@Test
+	public void testBlogPostsLinkExistsInContentSpace() {
+		ApioClientBuilder.given(
+		).basicAuth(
+			"test@liferay.com", "test"
+		).header(
+			"Accept", "application/hal+json"
+		).when(
+		).get(
+			_rootEndpointURL.toExternalForm()
+		).follow(
+			"_links.content-space.href"
+		).then(
+		).statusCode(
+			200
+		).body(
+			"_embedded.ContentSpace.find {it.name == '" +
+				BlogApioTestBundleActivator.CONTENT_SPACE_NAME +
+					"'}._links.blogPosts.href",
+			IsNull.notNullValue()
+		);
+	}
+
+	@Test
+	public void testCreateBlog() throws Exception {
+		String blogHref = ApioClientBuilder.given(
+		).basicAuth(
+			"test@liferay.com", "test"
+		).header(
+			"Accept", "application/hal+json"
+		).header(
+			"Content-Type", "application/json"
+		).body(
+			_read("test-create-blog-posting.json")
+		).when(
+		).post(
+			_blogPostsHref
+		).then(
+		).statusCode(
+			200
+		).body(
+			"dateCreated", IsNull.notNullValue()
+		).body(
+			"dateModified", IsNull.notNullValue()
+		).body(
+			"datePublished", IsNull.notNullValue()
+		).body(
+			"alternativeHeadline", IsEqual.equalTo("alternativeHeadline")
+		).body(
+			"articleBody", IsEqual.equalTo("articleBody")
+		).body(
+			"caption", IsEqual.equalTo("caption")
+		).body(
+			"description", IsEqual.equalTo("description")
+		).body(
+			"encodingFormat", IsEqual.equalTo("text/html")
+		).body(
+			"friendlyUrlPath", IsEqual.equalTo("friendlyurlpath")
+		).body(
+			"headline", IsEqual.equalTo("headline")
+		).body(
+			"keywords",
+			IsEqual.equalTo(Arrays.asList("keyword1", "keyword2", "keyword3"))
+		).extract(
+		).path(
+			"_links.self.href"
+		);
+
+		_deleteBlog(blogHref);
+	}
+
+	@Test
+	public void testDeleteBlog() throws Exception {
+		String blogHref = ApioClientBuilder.given(
+		).basicAuth(
+			"test@liferay.com", "test"
+		).header(
+			"Accept", "application/hal+json"
+		).header(
+			"Content-Type", "application/json"
+		).body(
+			_read("test-create-blog-posting.json")
+		).when(
+		).post(
+			_blogPostsHref
+		).then(
+		).extract(
+		).path(
+			"_links.self.href"
+		);
+
+		_deleteBlog(blogHref);
```

Comment:
It is in general not a good practice to do this with the _deleteBlog, when this test fails, all the code that you have here is  the one to obtain the blogHref. Looks like the _deleteBlog code is just for cleaning, no the test itself